devtools/maven/src/main/java/io/quarkus/maven/GenerateConfigMojo.java

```diff
@@ -95,48 +91,28 @@ public void execute() throws MojoExecutionException {
             getLog().info("Type of the artifact is POM, skipping generate-config goal");
             return;
         }
-
-        final Artifact projectArtifact = project.getArtifact();
-        final AppArtifact appArtifact = new AppArtifact(projectArtifact.getGroupId(), projectArtifact.getArtifactId(),
-                projectArtifact.getClassifier(), "pom",
-                projectArtifact.getVersion());
-        final AppModel appModel;
-        final BootstrapAppModelResolver modelResolver;
-        try {
-            LocalProject localProject = LocalProject.load(project.getBasedir().toPath());
-            modelResolver = new BootstrapAppModelResolver(
-                    MavenArtifactResolver.builder()
-                            .setRepositorySystem(repoSystem)
-                            .setRepositorySystemSession(repoSession)
-                            .setRemoteRepositories(repos)
-                            .setWorkspace(localProject.getWorkspace())
-                            .build());
-            appModel = modelResolver.resolveModel(appArtifact);
-        } catch (Exception e) {
-            throw new MojoExecutionException("Failed to resolve application model " + appArtifact + " dependencies", e);
-        }
         if (project.getResources().isEmpty()) {
             throw new MojoExecutionException("No resources directory, cannot create application.properties");
         }
+        try {
+            CuratedApplication curatedApplication = QuarkusBootstrap.builder(Paths.get(project.getBuild().getOutputDirectory()))
```

Comment:
Looks like `curatedApplication` isn't closed. I would also strongly recommend creating maven artifact resolvers in mojos and passing them to the bootstrap.

Reply:
how do you do that? This is a seperate JVM. 

One option could be to resolve the app model in the mojo and serialize the resolved app.

Reply:
Isn't curated application going to be usually created in the Mojo when running from maven?

Reply:
Except for dev mode, and tests (which have always done their own resolution). I can make dev mode work by serialising the result into the dev mode runner jar, but there is not much I can do with tests.